src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,380 @@
+        /// void DoSomething(int i, Derived1 d) { }
+        /// void DoSomething(string s, Derived2 d) { }
+        /// void DoSomething(int i, Derived3 d) { }
+        /// 
+        /// Base b;
+        /// DoSomething(1, [||]b);
+        ///
+        /// *void DoSomething(string s, Derived2 d) { }* is not the perfect match candidate function for
+        /// *DoSomething(1, [||]b)* because int and string are not ancestor-descendant relationship. Thus,
+        /// Derived2 is not a potential conversion type.
+        /// 
+        /// <param name="argumentList"> The argument list of invocation expression</param>
+        /// <param name="parameters"> The parameters of function</param>
+        /// <param name="targetArgument"> The target argument that contains target node</param>
+        /// <param name="targetArgumentConversionType"> Output the corresponding parameter type of
+        /// the target arugment if function returns true</param>
+        /// <returns>
+        /// True, if arguments and parameters match perfectly. <paramref name="targetArgumentConversionType"/> Output the corresponding parameter type
+        /// False, otherwise.
+        /// </returns>
+        private static bool CanArgumentTypesBeConvertedToParameterTypesTest(
+            SemanticModel semanticModel, SyntaxNode root, ArgumentListSyntax argumentList,
+            ImmutableArray<IParameterSymbol> parameters, ArgumentSyntax targetArgument,
+            CancellationToken cancellationToken, [NotNullWhen(true)] out ITypeSymbol? targetArgumentConversionType)
+        {
+            targetArgumentConversionType = null;
+
+            // No conversion happens under this case
+            if (parameters.Length == 0)
+                return false;
+
+            var arguments = argumentList.Arguments;
+            var newArguments = new List<ArgumentSyntax>();
+
+            for (var i = 0; i < arguments.Count; i++)
+            {
+                // Parameter index cannot out of its range, #arguments is larger than #parameter only if the last parameter with keyword params
+                var parameterIndex = Math.Min(i, parameters.Length - 1);
+
+                // If the argument has a name, get the corresponding parameter index
+                var nameSyntax = arguments[i].NameColon?.Name;
+                if (nameSyntax != null)
+                {
+                    var name = nameSyntax.Identifier.ValueText;
+                    if (!FindCorrespondingParameterByName(name, parameters, ref parameterIndex))
+                        return false;
+                }
+
+                // The argument is either in order with parameters, or have a matched name with parameters.
+                // "argumentType.Type" could be null when "augumentExpression" is lambada expression.
+                var augumentExpression = arguments[i].Expression;
+                var argumentType = semanticModel.GetTypeInfo(augumentExpression, cancellationToken);
```

Comment:
so you're grabbing the type of the argument in order to pass it to ClassifyCommonConversion.  however, CSharpSemanticModel has a .ClassifyConversion that takes an expression-node, and type.  That's the preferred form.  it will work for things like `null`.

Reply:
Just to confirm that `semanticModel.ClassifyConversion` is the one I invoked below, right? :)